tools/shared.py

```diff
@@ -1869,8 +1869,6 @@ def llvm_opt(filename, opts, out=None):
       inputs = [inputs]
     else:
       assert out, 'must provide out if llvm_opt on a list of inputs'
-    if type(opts) is int:
```

Comment:
i think we still need the inlining logic here, that is, it's ok to remove these two lines, but instead we should add
````
if not Building.can_inline():
  opts.append('-disable-inlining')
````
I'm surprised no test depends on that, we need to figure that out. Either we need a test, or I am misunderstanding something in the code.

Reply:
That's what I tried first but it causes failure on `other.test_llvm_lto`. Adding `not '-Oz'` prevents it but not sure why it happens.

Reply:
`parse_args()` code says that the optimization level `-Oz` sets `INLINING_LIMIT=25`, but `can_inline()` still gives false value while it should give true. It seems it should be investigated more.

Reply:
@kripken There was a test but was removed by this commit, do you remember why?

https://github.com/kripken/emscripten/commit/d6625999e9eb59c951a137f43a258a64ef44773d#diff-9ee3e07b8e34e901a62806f99766f262

Reply:
Hmm the test was removed in 2014. I'd guess an llvm inlining change made it no longer pass, and it's not an asm.js test anyhow.

Yeah, `INLINING_LIMIT` is how much to limit inlining, and 0 means no limit :)

But when you add those two lines, isn't it executing the same llvm commands as before? comparing those two should show what's going wrong here.

Reply:
No, those lines are currently only executed for only O1/O2/O3, not Os or Oz. (because `s` and `z` are not integer)

Reply:
Hmm, yes. And when you don't have those two lines, does it execute the same llvm commands as before both for Os and (for example) O2 as it did before this PR? I have the feeling something has changed here.

Reply:
Nothing disables the inlining for O1/O2/O3 so yes.

With -O2 (mapped to O3 automatically):

```
DEBUG:root:emcc: LLVM opts: -O3 -disable-loop-vectorization -disable-slp-vectorization -vectorize-loops=false -vectorize-slp=false -vectorize-slp-aggressive=false  [num inputs: 1]
```

With -Os:

```
DEBUG:root:emcc: LLVM opts: -Os -disable-loop-vectorization -disable-slp-vectorization -vectorize-loops=false -vectorize-slp=false -vectorize-slp-aggressive=false  [num inputs: 1]
```